pyodide_build/common.py

```diff
@@ -30,11 +32,48 @@
 
 
 def xbuildenv_dirname() -> str:
-    from pyodide_build import __version__
+    try:
+        from pyodide_build import __version__
+    except ImportError:
+        __version__ = "0.0.0"
 
     return f".pyodide-xbuildenv-{__version__}"
 
 
+@lru_cache(maxsize=1)
```

Comment:
I'd suggest using `functools.cache` instead for simplicity.
```suggestion
@cache
```